packages/airnode-node/src/api/index.ts

```diff
@@ -105,46 +101,49 @@ async function signWithTemplateId(
   );
 }
 
+type CallApiConfig = Pick<Config, 'ois' | 'apiCredentials'> &
+  Partial<Pick<Config, 'chains'>> &
+  Partial<{
+    nodeSettings: Pick<Config['nodeSettings'], 'airnodeWalletMnemonic'>;
+  }>;
+
 export interface CallApiPayload {
-  readonly config: Config;
+  readonly config: CallApiConfig;
   readonly aggregatedApiCall: AggregatedApiCall;
 }
```

Comment:
I think we should distinguish the payload (and config) types for different call types (regular vs. http vs. http signed data). It makes no sense to keep the fields there as optional when we know that only some call type needs those fields (or rather will contain them for certain).Similar to how we use the `AggregatedApiCall` type.

Reply:
Sounds good 👍🏻  Please let me know if [something like this](https://github.com/api3dao/airnode/pull/1308/commits/a4faf2aa8ac236c4ceffd0f06e0a8987aff2a98f) is what you were thinking of.

Reply:
Almost. I think moving the `type` property from `*AggregatedApiCall` to `*ApiCallPayload` will make more sense now that we need to distinguish both and the payload is superior (contains the aggregated API call). You'll get rid of a lot of those unnecessary `as` statements. I think it shouldn't cause any problems and should be doable but can't really be certain without trying it. @Siegrift you've worked on this part of the code before, what do you think?

Reply:
I even now wonder if the `type` property is even needed since we can always check using `instanceof` to determine the `AggregatedApiCall` type 🤔

Reply:
> I think moving the type property from *AggregatedApiCall to *ApiCallPayload will make more sense now that we need to distinguish both and the payload is superior (contains the aggregated API call)

Yeah, now that the config is also differentiated based on the type it makes sense. That said, I am not sure if introducing a "config.json subset type" is worth the complexity and whether we should not just use the "whole" config.json (but I don't remember what was the decision when we discussed this for the Airkeeper).

> I even now wonder if the type property is even needed since we can always check using instanceof to determine the AggregatedApiCall type

Not sure what you mean. `instanceof` can be used only to check prototype inheritance (so you mostly use it for class/insance check). You can't use that to check if some value is of a certain type (there is no such feature in TS since the types are removed at runtime).

Reply:
I just meant checking if an object has some shape or contains certain properties in order to determine they type but I'll just keep it as is.

Reply:
I've just applied the changes but ended up making changes in places I was expecting not to touch 🤭

Please review it again @amarthadan when you have the chance